azkaban-common/src/main/java/azkaban/executor/ExecutorHealthChecker.java

```diff
@@ -0,0 +1,158 @@
+/*
+* Copyright 2019 LinkedIn Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+package azkaban.executor;
+
+import azkaban.Constants.ConfigurationKeys;
+import azkaban.utils.Pair;
+import azkaban.utils.Props;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Periodically checks the health of executors. Finalizes flows or sends alert emails when needed.
+ */
+@SuppressWarnings("FutureReturnValueIgnored")
+@Singleton
+public class ExecutorHealthChecker {
+
+  private static final Logger logger = LoggerFactory.getLogger(ExecutorHealthChecker.class);
```

Comment:
Thanks for the reference. For references for logger, could you please remove "this."? It's a bit confusing, since it looks like an object reference: https://docs.oracle.com/javase/tutorial/java/javaOO/classvars.html.

Reply:
Will fix it, thanks!